zarnitsa/series/DataAugmenterTimeSeries.py

```diff
@@ -0,0 +1,159 @@
+import os
+from typing import Tuple
+
+from scipy.signal import lfilter, savgol_filter
+import numpy as np
+import pandas as pd
+
+from DataAugmenter import AbstractDataAugmenter
+
+
+class DataAugmenterTimeSeries(AbstractDataAugmenter):
+    def __init__(self, n_jobs=1):
+        self.n_jobs = n_jobs
+
+    def augment_dataframe(
+        self, df: pd.DataFrame, columns=None, **kwargs
+    ) -> pd.DataFrame:
+        """Augment dataframe data. Pandas dataframe"""
+        columns = columns if columns else df.columns
+        for col in columns:
+            df[col] = self.augment_column(df[col], **kwargs)
+        return df
+
+    def augment_column(
+        self,
+        col: pd.Series,
+        func: str,
+        **kwargs,
+    ) -> pd.Series:
+        """Augment dataframe data. Pandas dataframe"""
+        func = {
+            "dropping": self.data_droping,
+            "data_interpolation": self.data_interpolation,
+            "data_extrapolation": self.data_extrapolation,
+            "data_denoiser": self.data_denoiser,
+            "data_noiser": self.data_noiser,
+        }
+        return func[func_name](col, **kwargs)
+
+    def _prepare_data_to_aug(self, data, freq=0.2) -> Tuple[pd.Series, pd.Series]:
+        pass
+
+    @staticmethod
+    def data_droping(col, limit=None, freq=0.2, **kwargs) -> pd.Series:
+        """
+        Extrapolate pandas column
+        """
+        if len(col) == 0:
+            raise ValueError(f"Iterable object <{type(col)}> is empty! Check input!")
+        if limit:
+            freq = limit / col.shape[0]
+        drop_indices = col.sample(frac=freq).index
+        return col[[i for i in col.index if i not in drop_indices]]
+
+    @staticmethod
+    def data_interpolation(
+        col, method="linear", limit=None, freq=1.0, order=2, **kwargs
+    ) -> pd.Series:
+        """
+        Interpolate pandas column
+        filling the gaps
+        """
+        if len(col) == 0:
+            raise ValueError(f"Iterable object <{type(col)}> is empty! Check input!")
+
+        if limit is None:
+            limit = int(col[col.isna()].shape[0] * freq)
+
+        if method in ["fillna"]:
+            col = col.fillna(kwargs["value_to_fill"], limit=limit)
+        elif method in ["linear", "pad", "polynomial"]:
+            indices = col.index
+            col = col.reset_index(drop=True)
```

Comment:
it is not the best practice to redefine variables and it is undesirable to redefine arguments

Reply:
idk why is this here. just really useful ops